packages/core/src/lib/passkey/options.ts

```diff
@@ -0,0 +1,130 @@
+import type { Authenticator } from "../../types"
+import {
+  generateAuthenticationOptions,
+  generateRegistrationOptions,
+} from "@simplewebauthn/server"
+import type {
+  PublicKeyCredentialCreationOptionsJSON,
+  PublicKeyCredentialRequestOptionsJSON,
+} from "@simplewebauthn/server/script/deps"
+import {
+  assertAdapterImplementsMethods,
+  type Adapter,
+  type AdapterUser,
+  AdapterAccount,
+} from "../../adapters"
+import type { Options } from "./types"
+import { randomString } from "../web"
+import { MissingAdapter } from "../../errors"
+
+async function getUserAndAuthenticators(
+  options: Options,
+  email?: string
+): Promise<[Authenticator[] | undefined, AdapterUser | null, AdapterAccount | null]> {
+  const { adapter: _adapter, provider } = options
+  const adapter: Adapter | undefined = _adapter
+
+  // Validate that the adapter is defined and implements the required methods
+  if (!adapter)
+    throw new MissingAdapter(
+      "WebaAuthn getUserAndAuthenticators requires an adapter."
+    )
+  assertAdapterImplementsMethods(
+    "WebaAuthn getUserAndAuthenticators requires an adapter that implements",
```

Comment:
You're correct, but my mistake here was not setting `residentKey` to `required`.
I'll remove `requireResidentKey` since it only exists to support the previous spec.

We wan't `required` here because passkeys are always "resident keys" as [Discoverable Credentials](https://passkeys.dev/docs/reference/terms/#discoverable-credential)

Fixed in 2a224e30e67fa041f8093123364d233aaaf62351